projects/RabbitMQ.Client/client/impl/CommandAssembler.cs

```diff
@@ -162,16 +159,21 @@ private bool ParseHeaderFrame(in InboundFrame frame)
                 throw new MalformedFrameException(message: msg, canShutdownCleanly: false);
             }
 
-            _rentedHeaderArray = totalBodyBytes != 0 ? frame.TakeoverPayload() : Array.Empty<byte>();
```

Comment:
This set the rented array to Array.Empty, if the totalBodyBytes was 0, but the Memory itself was still set. So the code after continued to work as it reads off the Memory itself, but the backing array was already returned can could've been changed.

This change makes it so that if it's only the 14 bytes of the Header, we'll return it. (As the basic property is empty in this case), and otherwise slice it and take over the array.